classes/task/robot_cleanup.php

```diff
@@ -0,0 +1,71 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * local_linkchecker_robot
+ *
+ * @package    local_linkchecker_robot
+ * @copyright  2016 Suan Kan <suankan@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace local_linkchecker_robot\task;
+
+/**
+ * robot_cleanup
+ *
+ * @package    local_linkchecker_robot
+ * @copyright  2016 Suan Kan <suankan@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class robot_cleanup extends \core\task\scheduled_task {
+
+    /**
+     * Get task name
+     */
+    public function get_name() {
+        return get_string('robotcleanup', 'local_linkchecker_robot');
+    }
+
+    /**
+     * Finds and deletes all bad URLs which belong to previous scrape session, but not the current one.
+     *
+     * @param null $time. For unit tests purpose: we need to simulate execution of this task at any arbitrary time.
+     */
+    public function execute($time = null) {
+        global $DB;
+
+        if ($time) {
```

Comment:
I'd save a few lines and do

function execute($currenttime = null) {

if (!$currenttime){
  $currenttime = time();
}


But this is super minor stuff everything else looks great. The only other comment, which is minor, is that you have 8 records in the unit test and then remove 7, and end up with 1. The test would be equally robust with only 3 records with less code.